contribs/application/src/main/java/org/matsim/application/analysis/population/TripAnalysis.java

```diff
@@ -188,6 +188,45 @@ public Integer call() throws Exception {
 		return 0;
 	}
 
+	private Character detectDelimiter(String path) throws IOException {
+		// Create a map to count occurrences of potential delimiters
+		Map<Character, Integer> delimiterCounts = new HashMap<>();
```

Comment:
I would not create an extra map to store the occurrences. You can just create 3 variables for your delimiters and compare them directly.